src/components/interaction-designer/block-editors/GroupMembershipEditor.vue

```diff
@@ -1,153 +1,188 @@
+  }
+
+  get groupOptions(): unknown {
+    return this.availableGroups ?? this.customGroupOptions
+  }
 
-  MEMBERSHIP_ACTION = {
-    ADD: 'add',
-    SET_FROM_EXPRESSION: 'set_from_expression',
-    REMOVE: 'remove',
+  get membershipActions(): MembershipAction[] {
+    return [
+      {
+        id: 'addGroup',
+        localizationKey: 'flow-builder.set-group-membership',
+        value: MEMBERSHIP_ACTION.ADD,
+      },
+      {
+        id: 'removeGroup',
+        localizationKey: 'flow-builder.remove-group-membership',
+        value: MEMBERSHIP_ACTION.REMOVE,
+      },
+      {
+        id: 'clearGroup',
+        localizationKey: 'flow-builder.clear-group-membership',
+        value: MEMBERSHIP_ACTION.CLEAR,
+      },
+    ]
   }
 
-  get membershipAction(): string {
-    if (this.isMember === NULL_STRING_EXPRESSION) {
-      return this.MEMBERSHIP_ACTION.REMOVE
-    } else if (this.isMember === TRUTHY_STRING_EXPRESSION) {
-      return this.MEMBERSHIP_ACTION.ADD
+  get membershipAction(): MEMBERSHIP_ACTION {
+    const {clear, is_member: isMember} = this.block.config
+
+    if (clear === true) {
+      return MEMBERSHIP_ACTION.CLEAR
     }
-    return this.MEMBERSHIP_ACTION.SET_FROM_EXPRESSION
+
+    return isMember === true
+      ? MEMBERSHIP_ACTION.ADD
+      : MEMBERSHIP_ACTION.REMOVE
   }
 
-  set membershipAction(value: string) {
-    if (value === this.MEMBERSHIP_ACTION.REMOVE) {
-      this.updateIsMemberExpression(NULL_STRING_EXPRESSION)
-    } else if (value === this.MEMBERSHIP_ACTION.ADD) {
-      this.updateIsMemberExpression(TRUTHY_STRING_EXPRESSION)
+  set membershipAction(value: MEMBERSHIP_ACTION) {
+    if (value === MEMBERSHIP_ACTION.CLEAR) {
+      this.cachedGroupsSelection = this.selectedGroups
+      this.updateBlockConfig({
+        clear: true,
+        groups: undefined,
+        is_member: undefined,
+      })
     } else {
-      this.updateIsMemberExpression(EMPTY_STRING_EXPRESSION)
+      this.updateBlockConfig({
+        clear: undefined,
+        groups: this.cachedGroupsSelection,
+        is_member: value === MEMBERSHIP_ACTION.ADD,
+      })
     }
   }
 
-  get groupKey(): string {
-    return get(this.block.config, 'group_key')
+  get isGroupListVisible(): boolean {
+    return this.membershipAction !== MEMBERSHIP_ACTION.CLEAR
   }
 
-  set groupKey(value: string) {
-    this.block_updateConfigByPath({
-      blockId: this.block.uuid,
-      path: 'group_key',
-      value,
+  get selectedGroups(): IGroupMembership[] {
+    return this.block.config.groups ?? []
+  }
+
+  set selectedGroups(groups: IGroupMembership[]) {
+    this.updateBlockConfig({
+      groups,
     })
   }
 
-  get isMember(): string {
-    return get(this.block.config, 'is_member', EMPTY_STRING_EXPRESSION)
+  updateBlockConfig(config: Partial<ISetGroupMembershipBlockConfig>): void {
+    Object.entries(config)
+      .forEach(([key, value]) => {
+        if (value === undefined) {
+          this.block_removeConfigByKey({
```

Comment:
We do have `additionalProperties: false` in the generated schemas, so `undefined` results in _Property … has not been defined and the schema does not allow additional properties_

Reply:
@object-required , hum ... I think that will complicate things in the future:
Let's say we import a flow having `{clear: true, is_member: false, groups: undefined}` , this will throw a validation error then; while this is not an error if we check the [proposal](https://github.com/FLOIP/flow-spec/issues/63) saying
> Add the "clear" field to the block config, which, if present and true, removes the contact from all groups. When the "clear" field is present, the other fields will be ignored.

In that case, I think we should have allowed such values in flow-runner [PR](https://github.com/FLOIP/flow-runner/pull/74/files#diff-ddd640bcf80ccacb518a2e6d05a01cfa42ed9301e7918bec9079fe2a59b28c19R13). What do you think?

I'm thinking about updating the flow-runner interface if you agree:
```
/**
 * Adding or removing contact from group(s)
 */
export interface IAddOrRemoveGroupMembership extends IBlockConfig {
  is_member: boolean
  groups: IGroupMembership[]
  clear?: false
}

/**
 * Removing contact from all groups
 * When the "clear" field is present and TRUE, the other fields will be ignored.
 */
export interface IClearGroupMembership extends IBlockConfig {
  is_member?: boolean
  groups?: IGroupMembership[]
  clear: true
}

export type ISetGroupMembershipBlockConfig = IAddOrRemoveGroupMembership | IClearGroupMembership
```

And that will simplify many things in the UI, like the usage of block_removeConfigByKey